Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -52,9 +47,7 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
     if (self) {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSMutableDictionary alloc] init];
-        self.reverseLookup                          =
-            [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                  valueOptions:NSMapTableWeakMemory];
+        self.reverseLookup                          = [[NSMutableDictionary alloc] init];
```

Comment:
The new expected behavior is that the view controller owns the cache and cleans it. The hash method on NSObject seems to suffice.